scripts/dep_update.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash -xe
+# Run pre and run hooks
+# Detect whether the repo has been modified
+# Create pr if required
+# Run post hook
+run_hook(){
+  if [[ -x "$1" ]]
+  then
+    $1 || exit 1
+  else
+    if [[ "$1" =~ .*/run ]]
+    then
+      echo "Run hook $1 not found."
+      exit 1
+    fi
+  fi
+}
+cd ${WORKSPACE}/repo
+run_hook "gating/update_dependencies/pre"
+run_hook "gating/update_dependencies/run"
+if [[ -z "$(git status -s)" ]]
+then
+  echo "Repo is clean, prep script made no changes to be committed."
+else
+  echo "Repo is dirty, preparing to propose changes."
+  # split URL into array separated by '/'s
+  IFS='/ ' read -r -a url_split <<< "$URL"
+  # owner/org is whatever is between the second and third / in the url
+  owner="${url_split[3]}"
+  # repo is anything after owner/
+  repo=${URL#https://github.com/${owner}/}
+  ssh_url="git@github.com:${owner}/${repo}"
+  message="This change updates the branch dependencies with the repository
+modifications made by running the Jenkins hook in
+gating/update_dependencies"
+
+  # Get / Create Jira issue
+  echo "Looking for Jira issue, will create if not found."
+  jira_summary="Update ${repo}:${BRANCH} dependencies"
+  issue=$(python ${WORKSPACE}/rpc-gating/scripts/jirautils.py \
+        --user "${JIRA_USER}" \
+        --password "${JIRA_PASS}" \
+        create_issue \
+          --project "${JIRA_PROJECT_KEY}" \
+          --summary "${jira_summary}" \
+          --description "${message}" \
+          --type "${JIRA_ISSUE_TYPE}" \
+          --label RE_DEP_UPDATE \
+          --existing-issue-query "project = ${JIRA_PROJECT_KEY} AND resolution = Unresolved AND summary ~ \"${jira_summary}\""
+  )
+  echo "Issue: ${issue}"
+
+  echo "Committing changes"
+  pr_branch="${issue}_${BRANCH}_dep_update"
+  title="${issue} Update ${BRANCH} dependencies"
+  git commit -a -m "${title}
+
+  ${message}"
+
+  echo "Pushing changes to repo branch"
+  git push -f "$ssh_url" "HEAD:${pr_branch}"
```

Comment:
Jenkins should maintain a fork for creating pull requests. RPCO has chosen to move away from creating pull request branches directly on rcbops/rpc-openstack, it would make sense to be consistent with that approach. Also given the permissions Jenkins has it probably reduces the risk of error.

Reply:
That's probably a good idea, although I'd vote for that as an improvement in another card.

Reply:
I think it woud make sense to make an exception to that rule for Jenkins. Automated branches are namespaced and will not conflict with well known branch names as has happened recently.

If the consensus is that Jenkins must fork, then I agree with @odyssey4me that should be another card. We'll have to auto create forks and update pr creation code to create cross repo PRs.